src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -630,6 +635,11 @@ namespace winrt::TerminalApp::implementation
 
         if (auto page{ weakThis.get() })
         {
+            page->_processingCommandlineArgs = true;
+            auto finishProcessing = wil::scope_exit([&page]() {
+                page->_processingCommandlineArgs = false;
```

Comment:
.... Can't you just use `_startupState <= StartupState::Initialized` instead of introducing a new member?

Reply:
No because we're trying to suppress notifications when processing commandline args regardless of initialization step. So your proposal works for wt.exe args from Run, but not from commandline mode in the command palette.